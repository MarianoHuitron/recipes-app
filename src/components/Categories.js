import React, { useEffect, useRef, useState } from 'react'
import { simpleFetch } from '../helper/fetch';
import CategoryItem from './CategoryItem';
import Loader from './Loader';

const Categories = ({ categorySelected, setCategorySelected }) => {

    const [categories, setCategories] = useState([]);
    const divScroll = useRef();

    useEffect(() => {
        const getCategories = async () => {
            const resp = await simpleFetch('/categories.php');
            setCategories(resp.categories);
        }
        getCategories();
    }, [setCategories]);

    const scroll = (e) => {
        const target = document.getElementById('horizontalDiv');
        const toLeft = e.deltaY < 0 && target.scrollLeft > 0;
        const toRight = e.deltaY > 0 && target.scrollLeft < target.scrollWidth - target.clientHeight;
        
        if(toLeft || toRight) {
            target.scrollLeft += e.deltaY;
        }
    }
   
    if(categories.length === 0) {
        return (
            <Loader />
        )
    }

    return (
        <div 
            className="mt-4 d-inline-flex overflow-auto scroll-horizontal" 
            style={{width: '100%'}}
            id="horizontalDiv"
            ref={ divScroll }
            onWheel={ scroll }           
        >
            {categories.map(category => (
                <CategoryItem 
                    key={ category.idCategory }
                    category={ category }
                    categorySelected={ categorySelected }
                    setCategorySelected={ setCategorySelected }
                />
            ))}
        </div>
    )
}

export default Categories
