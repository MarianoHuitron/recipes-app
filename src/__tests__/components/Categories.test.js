import { shallow } from 'enzyme';
import React from 'react';
import Categories from '../../components/Categories';

const setCategorySelected = jest.fn();
const categorySelected = 'Seafood';

describe('Testing <Categories />', () => {

    // let useEffect;

    // const mockUseEffect = () => {
    //     useEffect.mockImplementationOnce(f => f());
    // }

    // beforeEach(() => {
    //     useEff
    // })
    
    test('should render component', () => {
        const wrapper = shallow(
            <Categories 
                categorySelected={ categorySelected }
                setCategorySelected={ setCategorySelected }
            />
            
        );

        expect( wrapper ).toMatchSnapshot();
    })
    

})
