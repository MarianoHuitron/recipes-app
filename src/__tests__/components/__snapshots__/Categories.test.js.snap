// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Categories /> should render component 1`] = `<Loader />`;
